mars/deploy/oscar/tests/test_ray_dag.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2021 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import os
+
+import pytest
+
+from ....tests.core import DICT_NOT_EMPTY, require_ray
+from ....utils import lazy_import
+from ..local import new_cluster
+from ..tests import test_local
+from .modules.utils import (  # noqa: F401; pylint: disable=unused-variable
+    cleanup_third_party_modules_output,
+    get_output_filenames,
+)
+
+ray = lazy_import("ray")
+
+CONFIG_TEST_FILE = os.path.join(
+    os.path.dirname(__file__), "local_test_with_ray_dag_config.yml"
+)
+
+EXPECT_PROFILING_STRUCTURE = {
+    "supervisor": {
+        "general": {
+            "optimize": 0.0005879402160644531,
+            "stage_*": {
+                "tile(*)": 0.008243083953857422,
+                "gen_subtask_graph(*)": 0.012202978134155273,
+                "run": 0.27870702743530273,
+                "total": 0.30318617820739746,
+            },
+            "total": 0.30951380729675293,
+        },
+        "serialization": {},
+        "most_calls": DICT_NOT_EMPTY,
+        "slow_calls": DICT_NOT_EMPTY,
+        "band_subtasks": {},
+        "slow_subtasks": {},
+    }
+}
+EXPECT_PROFILING_STRUCTURE_NO_SLOW = copy.deepcopy(EXPECT_PROFILING_STRUCTURE)
+EXPECT_PROFILING_STRUCTURE_NO_SLOW["supervisor"]["slow_calls"] = {}
+
+
+@pytest.mark.parametrize(indirect=True)
+@pytest.fixture
+async def create_cluster(request):
+    start_method = os.environ.get("POOL_START_METHOD", None)
+    client = await new_cluster(
+        subprocess_start_method=start_method,
+        config=CONFIG_TEST_FILE,
```

Comment:
You can pass config directly without the config file:
```
    client = await new_cluster(
        "test_cluster",
        worker_num=5,
        worker_cpu=2,
        worker_mem=512 * 1024**2,
        supervisor_mem=100 * 1024**2,
        config={
            # coloring based fusion will make subtask too heterogeneous such that the speculative scheduler can't
            # get enough homogeneous subtasks to calculate statistics
            "task": {"default_config": {"fuse_enabled": False}},
            "scheduling": {
                "speculation": {
                    "enabled": True,
                    "interval": 0.5,
                    "threshold": 0.2,
                    "min_task_runtime": 2,
                    "multiplier": 1.5,
                },
                # used to kill hanged subtask to release slot.
                "subtask_cancel_timeout": 0.1,
            },
        },
    )
```

Reply:
There is a merge config logic in ray.new_cluster. A simple `{"task": {"task_executor_config": {"backend": "ray"}}}` config does not work for mars new_cluster. Maybe we can add the  merge config logic for mars in the future, but it's out of this PR because the PR is very large now.